src/main/java/org/openrewrite/java/testing/junitassertj/AssertNotNullToAssertThat.java

```diff
@@ -0,0 +1,141 @@
+            .declaringClass(ASSERTJ_QUALIFIED_ASSERTIONS_CLASS_NAME)
+            .flags(Flag.Public, Flag.Static)
+            .returnType("org.assertj.core.api.AbstractBooleanAssert")
+            .name(ASSERTJ_ASSERT_THAT_METHOD_NAME)
+            .parameter(JavaType.Primitive.Boolean, "arg1")
+            .build();
+
+    @Override
+    public J.CompilationUnit visitCompilationUnit(J.CompilationUnit cu) {
+        maybeRemoveImport(JUNIT_QUALIFIED_ASSERTIONS_CLASS);
+        return super.visitCompilationUnit(cu);
+    }
+
+    @Override
+    public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method) {
+        J.MethodInvocation original = super.visitMethodInvocation(method);
+        if (!JUNIT_ASSERT_TRUE_MATCHER.matches(method)) {
+            return original;
+        }
+
+        List<Expression> originalArgs = original.getArgs().getArgs();
+        Expression objToCheck = originalArgs.get(0);
+        Expression message = originalArgs.size() == 2 ? originalArgs.get(1):null;
+
+        // This creates the `assertThat(<EXPRESSION>)` method invocation. Without the type information for this invocation,
+        // the AddImport visitor won't add a static import for "assertThat" (assuming it doesn't already exist)
+        // because it won't be able to find a reference to the type.
+        J.MethodInvocation assertSelect = new J.MethodInvocation(
+                randomId(),
+                null,
+                null,
+                J.Ident.build(randomId(), ASSERTJ_ASSERT_THAT_METHOD_NAME, JavaType.Primitive.Void, EMPTY),
+                new J.MethodInvocation.Arguments(
+                        randomId(),
+                        Collections.singletonList(objToCheck),
+                        EMPTY
+                ),
+                assertThatMethodType,
+                EMPTY
+        );
+
+        // If the assertNotNull is the two-argument variant, we need to maintain the message via a chained method
+        // call to "as"/"withFailMessage". The message may be a String or Supplier<String>.
+        if (message != null) {
+            // In assertJ the "as" method has a more informative error message, but doesn't accept String suppliers
+            // so we're using "as" if the message is a string and "withFailMessage" if it is a supplier.
+            String messageAs = TypeUtils.isString(message.getType())? "as" : "withFailMessage";
+
+            assertSelect = new J.MethodInvocation(
+                    randomId(),
+                    assertSelect, // assertThat is the select for this method.
+                    null,
+                    J.Ident.build(randomId(), messageAs, null, EMPTY),
+                    new J.MethodInvocation.Arguments(
+                            randomId(),
+                            Collections.singletonList(message.withPrefix("")),
+                            EMPTY
+                    ),
+                    null,
```

Comment:
If it's not too much trouble, it doesn't hurt to have type attributions everywhere you can. Then this visitor could potentially compose with other visitors in the same pipeline that might want to do something with the "as" method. I agree this one isn't critical, though.

Reply:
Currently that MethodInvocation could be either an "as" method or a "withFailMessage," so adding type attribution would be more complicated. I recommend that we defer this change to after assertJ releases the new "as" method that accepts a String supplier since we'll be needing to refactor this code at that point anyway.